documentation/gnmi/design/sonic-dash.yang

```diff
@@ -0,0 +1,490 @@
+                }
+
+                leaf-list vni {
+                    type  uint32 {
+                        range 0..16777215;
+                    }
+                }
+
+            } /* end of list DASH_APPLIANCE_LIST */
+        } /* end of container DASH_APPLIANCE */
+
+        container DASH_ROUTING_TYPE {
+            description "DASH Routing types and actions; part of APP_DB";            
+
+            list DASH_ROUTING_TYPE_LIST {
+                key "name";
+
+                leaf name {
+                    type string {
+                        pattern "direct|vnet|vnet-direct|drop|appliance|privatelink|privatelinknsg|servicetunnel";
+                    }
+                }
+
+                list actions {
+                    key "name"
+ 
+                    leaf name {
+                        type string {
+                            length 1..255;
+                        }
+                    }
+
+                    leaf action_type {
+                        type string {
+                            pattern "none|maprouting|direct|staticencap|appliance|4to6|mapdecap|decap|drop"
+                        }
+                    }
+
+                    leaf encap_type {
+                        type string {
+                            pattern "vxlan|nvgre"
+                        }
+                    }
+
+                    leaf vni {
+                        type  uint32 {
+                            range 0..16777215;
+                        }
+                    }
+                }
+
+            } /* end of list DASH_ROUTING_TYPE_LIST */
+        } /* end of container DASH_ROUTING_TYPE */
+
+        container DASH_ROUTE_TABLE {
+            description "DASH Route table associated to an ENI; part of APP_DB";            
+
+            list DASH_ROUTE_TABLE_LIST {
+                key "eni prefix";
+
+                leaf eni {
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_ENI/dash:DASH_ENI_LIST/dash:name;
+                    }
+                }
+
+                leaf prefix {
+                     type union {
+                         type stypes:sonic-ip4-prefix;
+                         type stypes:sonic-ip6-prefix;
+                     }
+                }
+ 
+                leaf ip_version {
+                    type string {
+                        pattern "IPv4|IPv6"
+                    }
+                }
+
+                leaf routing_type {
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_ROUTING_TYPE/dash:DASH_ROUTING_TYPE_LIST/dash:name;
+                    }
+                }
+
+                leaf vnet {
+                    when "((current()/../routing_type = 'vnet') or (current()/../routing_type = 'vnet-direct'))";
+                    type leafref {
+                        path /dash:sonic-dash/dash:DASH_VNET/dash:DASH_VNET_LIST/dash:name;
```

Comment:
How is DASH_MAPPING_TABLE used for routing type vnet and vnet-direct? please add a description.